### DISCUSSION TITLE
forceInsert similar to forceUpdate?

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#2999] [General]
We have a scenario where we think a 'forceInsert' option in our Csla objects would make life easier (for us : )).
We are getting data for our object graph, in DataPortal.Fetch,  but in our current UI (convoluted and strange UI but users like it) users might decide to create a new record.
We have used object cloning in the past where we would mark the object new and set primary key to zero.

Kind Regards

**Comment by michaelcsikos** [comment]
It sounds like you're describing a **Create new copy** scenario. I would normally create a new object and pass in `Widget copyFrom` to the DataPortal Create method. Run down the object graph to copy all the values necessary. All objects will be new. You can allow the user to make edits if necessary, and then save as normal.
